src/node_crypto.cc

```diff
@@ -2789,26 +2791,30 @@ bool CipherBase::InitAuthenticated(const char* cipher_type, int iv_len,
 
 #ifdef NODE_FIPS_MODE
     // TODO(tniessen) Support CCM decryption in FIPS mode
-    if (kind_ == kDecipher && FIPS_mode()) {
+    if (mode == EVP_CIPH_CCM_MODE && kind_ == kDecipher && FIPS_mode()) {
       env()->ThrowError("CCM decryption not supported in FIPS mode");
       return false;
     }
 #endif
 
-    if (!EVP_CIPHER_CTX_ctrl(ctx_.get(), EVP_CTRL_CCM_SET_TAG, auth_tag_len,
+    // Tell OpenSSL about the desired length.
+    if (!EVP_CIPHER_CTX_ctrl(ctx_.get(), EVP_CTRL_AEAD_SET_TAG, auth_tag_len,
                              nullptr)) {
       env()->ThrowError("Invalid authentication tag length");
       return false;
     }
 
+    // Remember the given authentication tag length for later.
     auth_tag_len_ = auth_tag_len;
 
-    // The message length is restricted to 2 ^ (8 * (15 - iv_len)) - 1 bytes.
-    CHECK(iv_len >= 7 && iv_len <= 13);
-    if (iv_len >= static_cast<int>(15.5 - log2(INT_MAX + 1.) / 8)) {
-      max_message_size_ = (1 << (8 * (15 - iv_len))) - 1;
-    } else {
-      max_message_size_ = INT_MAX;
+    if (mode == EVP_CIPH_CCM_MODE) {
+      // The message length is restricted to 2 ^ (8 * (15 - iv_len)) - 1 bytes.
+      CHECK(iv_len >= 7 && iv_len <= 13);
+      if (iv_len >= static_cast<int>(15.5 - log2(INT_MAX + 1.) / 8)) {
+        max_message_size_ = (1 << (8 * (15 - iv_len))) - 1;
+      } else {
+        max_message_size_ = INT_MAX;
+      }
```

Comment:
This will conflict with #21462 when it lands (but no doubt you knew that since you reviewed it. :-))

Reply:
Yes, but thanks for pointing it out 😃